services/rds/src/main/java/org/finra/gatekeeper/services/auth/GatekeeperRoleService.java

```diff
@@ -181,4 +227,60 @@ private GatekeeperRdsRole checkGatekeeperRdsRole() {
             return GatekeeperRdsRole.UNAUTHORIZED;
         }
     }
+
+    public Map<String, AppSpecificApprovalThreshold> getApprovalPolicy(Map<String, RoleMembership> roleMemberships) {
+        Map<String, AppSpecificApprovalThreshold> approvalPolicy = new HashMap<>();
+        for(String application : roleMemberships.keySet()) {
+            approvalPolicy.put(application, getApprovalPolicyByApplication(roleMemberships.get(application)));
+        }
+        return approvalPolicy;
+    }
+
+    private AppSpecificApprovalThreshold getApprovalPolicyByApplication(RoleMembership roleMembership) {
+        gatekeeperRoles = new ArrayList<>();
+
+        gatekeeperRoles.addAll(Arrays.asList(RoleType.values()));
+
+        sdlcs = new ArrayList<>();
+        sdlcs.addAll(gatekeeperApprovalProperties.getDev().get("readonly").keySet());
```

Comment:
you can combine these into 1 line..
```suggestion
    sdlcs = new ArrayList<>(gatekeeperApprovalProperties.getDev().get("readonly").keySet())
```